### ISSUE TITLE
Add the ability to override the basic methods

### ISSUE THREAD

**Issue by mitiko** [Mitiko/SimpleNeuralNetwork#5]
This would mean my other repository Quadratic Neural Networks and this one would merge together.  
This could be if one could _override_ certain or all functions in the `Network` class without changing the main structure.  
A way to implement this is to parameterize certain basic actions in (for example) the `Network` class - like the weight change, based on the learning rate, error and inexplicitly other variables, derived from the `Network`

**Comment by mitiko** [comment]
I have decided to make another library for development of more complex networks
